### PR TITLE
[Need feedback]: Implemented isEqualToDictionary

### PR DESCRIPTION
**Pull request by KaiCode2**
Implemented isEqualToDictionary and would like some feedback.

### CODE REVIEW

**Comment by amorde** [comment]
:+1:

File: Foundation/NSDictionary.swift [hunk 1]
```diff
@@ -278,7 +278,19 @@ public class NSDictionary : NSObject, NSCopying, NSMutableCopying, NSSecureCodin
     public func descriptionWithLocale(locale: AnyObject?, indent level: Int) -> String { NSUnimplemented() }
     
     public func isEqualToDictionary(otherDictionary: [NSObject : AnyObject]) -> Bool {
-        NSUnimplemented()
+        if count != otherDictionary.count {
+            return false
+        }
+        
+        for (key, value) in _storage {
```
Comment (gribozavr) [inline Foundation/NSDictionary.swift]: Is NSDictionary subclassable?  Then iterating over _storage wouldn't use the data from the subclass.

**Comment by gribozavr** [comment]
It would be great if you included tests.

File: Foundation/NSDictionary.swift (hunk 1 above)
Comment (amorde) [inline Foundation/NSDictionary.swift]: I think the solution here would be to use `keyEnumerator()`

Comment (KaiCode2) [inline Foundation/NSDictionary.swift]: @gribozavr Just corrected this issue using @amorde suggestion.

**Comment by KaiCode2** [comment]
> It would be great if you included tests.

Absolutely, I'll incorporate the tests now 😊

**Comment by phausler** [comment]
This does not handle the CF bridged dictionary case or subclasses; _storage is only available when it is a NSDictionary constructed from swift and not a custom subclass etc.

**Comment by phausler** [comment]
Looks good to me; a great follow up for this would definitely be some unit tests. However since it looks pretty cut and dry merging it in.

**Comment by KaiCode2** [comment]
@phausler O, thanks 😊

**Pull request merged**